examples/connection_snippets.rs

```diff
@@ -0,0 +1,38 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+extern crate mongodb;
+
+use mongodb::error::Result;
+
+const CONNECTION_URI: &str =
+    "mongodb+srv://<username>:<password>@<cluster-url>/<dbname>?w=majority";
+
+#[cfg(feature = "sync")]
+fn main() -> Result<()> {
+    use mongodb::sync::Client;
+    let client = Client::with_uri_str(CONNECTION_URI)?;
+    let database = client.database("test");
+    Ok(())
+}
+
+#[cfg(feature = "tokio-runtime")]
+#[tokio::main]
+async fn main() -> Result<()> {
```

Comment:
for Swift they requested that we only do the async API, so we probably want to do the same here. And we should probably just stick to tokio to keep things simple.

Reply:
Maybe we can split each feature into a separate example file?

Reply:
That was my first approach, but in order for the examples to compile there needs to be a `main` function present for each feature, since these examples need to be able to be run. Is there a way to enable/disable an entire file based on features?

Reply:
I suppose I could also just include an empty main for the compiler's sake

Reply:
I think the main function itself should be identical for tokio and async-std, right? The only difference will be using `tokio::main` versus `async_std::main`. Maybe we could cheat a little and conditionally import whichever `main` we need and then define a single async main:

```rust
#[cfg(feature = "tokio-runtime")]
use tokio::main;

#[cfg(feature = "async-std-runtime")]
use async_std::main;

#[cfg(not(feature = "sync"))]
#[main]
async fn main() -> Result<()> {
   ...
}
```